src/packages/core/src/utils/create-or-update-entities.ts

```diff
@@ -125,32 +127,69 @@ export const createOrUpdateEntities = async <G extends WithId, D extends BaseDat
 			const isRelatedEntity = relatedEntity && relatedEntity.prototype instanceof GraphQLEntity;
 
 			if (isRelatedEntity) {
-				// We have a related entity so lets check if we need to do an update or create
-				const result = isLinking(childNode) // if all entities are in the format {id: ""} then we are just linking the entity
-					? childNode
-					: await callChildMutation(
-							getMutationName(relatedEntity.name, childNode),
-							childNode,
-							info,
-							context
-					  );
-				node = {
-					...node,
-					[key]: result,
-				};
+				if (isLinking(childNode)) {
+					// If it's a linking entity or an array of linking entities, nothing to do here
+				} else if (Array.isArray(childNode)) {
+					// If we have an array, we may need to create the parent first as children need reference to the parent
+
+					// As are updating the parent from the child, we can remove this key
+					delete node[key as keyof Partial<G>];
```

Comment:
I don't totally understand this. Are we deleting this item so that it doesnt appear in the next iteration of the loop?

Reply:
The connection is in reverse when we have an array of children. So...

Artist has albums.

an Album has an artistId in the database. 

Therefore, we don't need to send the `albums[]` array when we create the Artist.